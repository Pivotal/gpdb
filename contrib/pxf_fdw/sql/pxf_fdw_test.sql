-- start_matchsubs
--
-- # create a match/subs expression
--
-- m/ERROR:.*invalid foreign table option\. The \'protocol\' option cannot be defined for PXF foreign tables.*/
-- s/ERROR:.*invalid foreign table option\. The \'protocol\' option cannot be defined for PXF foreign tables.*/ERROR:  invalid foreign table option. The 'protocol' option cannot be defined for PXF foreign tables/
--
-- end_matchsubs

CREATE EXTENSION IF NOT EXISTS pxf_fdw;

CREATE SERVER demo
    FOREIGN DATA WRAPPER pxf_fdw;

-- Foreign table creation fails if `protocol` option is provided
CREATE FOREIGN TABLE invalid_protocol_option( login_time interval )
    SERVER demo
    OPTIONS ( protocol 'hdfs' );

CREATE ROLE pxf_fdw_user;

-- User mapping creation fails if `protocol` option is provided
CREATE USER MAPPING FOR pxf_fdw_user
    SERVER demo
    OPTIONS ( protocol 'hdfs' );

-- Succeeds when protocol is not provided
CREATE USER MAPPING FOR pxf_fdw_user
    SERVER demo;

-- Altering a user mapping  fails if `protocol` option is added
ALTER USER MAPPING FOR pxf_fdw_user
    SERVER demo
    OPTIONS ( protocol 'hdfs' );

CREATE FOREIGN TABLE pxf_foreign_table( login_time interval )
    SERVER demo;

-- Altering a foreign table fails if `protocol` option is added
ALTER FOREIGN TABLE pxf_foreign_table
    OPTIONS (ADD protocol 'hdfs');

DROP USER MAPPING FOR pxf_fdw_user SERVER demo;
DROP FOREIGN TABLE pxf_foreign_table;
DROP SERVER demo;
