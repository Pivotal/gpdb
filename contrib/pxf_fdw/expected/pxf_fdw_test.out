-- start_matchsubs
--
-- # create a match/subs expression
--
-- m/ERROR:.*invalid foreign table option\. The \'protocol\' option cannot be defined for PXF foreign tables.*/
-- s/ERROR:.*invalid foreign table option\. The \'protocol\' option cannot be defined for PXF foreign tables.*/ERROR:  invalid foreign table option. The 'protocol' option cannot be defined for PXF foreign tables/
--
-- end_matchsubs
CREATE EXTENSION IF NOT EXISTS pxf_fdw;
CREATE SERVER demo
    FOREIGN DATA WRAPPER pxf_fdw;
-- Foreign table creation fails if `protocol` option is provided
CREATE FOREIGN TABLE invalid_protocol_option( login_time interval )
    SERVER demo
    OPTIONS ( protocol 'hdfs' );
ERROR:  invalid foreign table option. The 'protocol' option cannot be defined for PXF foreign tables
CREATE ROLE pxf_fdw_user;
NOTICE:  resource queue required -- using default resource queue "pg_default"
-- User mapping creation fails if `protocol` option is provided
CREATE USER MAPPING FOR pxf_fdw_user
    SERVER demo
    OPTIONS ( protocol 'hdfs' );
ERROR:  invalid user mapping option. The 'protocol' option cannot be defined for PXF user mappings
-- Succeeds when protocol is not provided
CREATE USER MAPPING FOR pxf_fdw_user
    SERVER demo;
-- Altering a user mapping  fails if `protocol` option is added
ALTER USER MAPPING FOR pxf_fdw_user
    SERVER demo
    OPTIONS ( protocol 'hdfs' );
ERROR:  invalid user mapping option. The 'protocol' option cannot be defined for PXF user mappings
CREATE FOREIGN TABLE pxf_foreign_table( login_time interval )
    SERVER demo;
-- Altering a foreign table fails if `protocol` option is added
ALTER FOREIGN TABLE pxf_foreign_table
    OPTIONS (ADD protocol 'hdfs');
ERROR:  invalid foreign table option. The 'protocol' option cannot be defined for PXF foreign tables
DROP USER MAPPING FOR pxf_fdw_user SERVER demo;
DROP FOREIGN TABLE pxf_foreign_table;
DROP SERVER demo;
